src/components/Toast/README.md

```diff
@@ -166,32 +161,25 @@ Action should:
 Use to convey general confirmation or actions that aren’t critical. For example, you might show a toast message to inform the merchant that their recent action was successful.
 
 ```jsx
-class ToastExample extends React.Component {
-  state = {
-    showToast: false,
-  };
-
-  render() {
-    const {showToast} = this.state;
-    const toastMarkup = showToast ? (
-      <Toast content="Message sent" onDismiss={this.toggleToast} />
-    ) : null;
-
-    return (
-      <div style={{height: '250px'}}>
-        <Frame>
-          <Page title="Toast example">
-            <Button onClick={this.toggleToast}>Show Toast</Button>
-            {toastMarkup}
-          </Page>
-        </Frame>
-      </div>
-    );
-  }
-
-  toggleToast = () => {
-    this.setState(({showToast}) => ({showToast: !showToast}));
-  };
+function ToastExample() {
+  const [active, setActive] = useState(false);
+
+  const toggleActive = useCallback(() => setActive(!active), [active]);
```

Comment:
```suggestion
  const toggleActive = useCallback(() => setActive((state) => !state), []);
```

Reply:
For my understanding, how does this not modify all of the values in state if we only want to change `active`?

Reply:
For some reason, I assumed `set*` didn't accept functions or callbacks. Guess I was wrong about the former 😄 I'll comb through my other examples and update them where it's needed.

> For my understanding, how does this not modify all of the values in state if we only want to change active?

When using hooks your able to have a single state opposed to state objects in class-based components. If you do decide to use an object to hold state using `useState`, you won't be able to partially update state as you can in class base components.

```tsx
const [booleanValue, setBooleanValue] = useState(false)
const toggleBooleanValue = useCallback(() => setActive(booleanValue => !booleanValue), []);
```

Reply:
Crosslinking ya to the other PR where I mentioned this, which has some explanations and links to docs https://github.com/Shopify/polaris-react/pull/2126#discussion_r323986229